lib/python/ray/worker.py

```diff
@@ -511,13 +610,17 @@ def process_task(task): # wrapping these lines in a function should cause the lo
     else:
       store_outputs_in_objstore(return_objrefs, outputs, worker) # store output in local object store
       ray.lib.notify_task_completed(worker.handle, True, "") # notify the scheduler that the task completed successfully
+    finally:
+      # Reinitialize the values of reusable variables that were used in the task
+      # above so that changes made to their state do not affect other tasks.
+      ray.reusables._reinitialize()
   while True:
-    (task, function) = ray.lib.wait_for_next_message(worker.handle)
+    (task, function, variable_info) = ray.lib.wait_for_next_message(worker.handle)
```

Comment:
e.g. `reusable_variable` or `reusable_variable_definition`. I'd prefer the latter since it's more descriptive, but to make it parallel with the other tuple members you might also change `function` to `function_definition` or something.